### PR TITLE
Add transactional start slot tracking and unit tests for bids service

### PR DESCRIPTION
**Pull request by xav**
## Description

This PR adds proper transactional handling for start slot metadata updates in the bids service and comprehensive unit test coverage.

## Changes

- Implement transactional start slot metadata updates with per-provider slot tracking
- Replace simple slot assignment with `BeginTx/SetMetadata/CommitTx` pattern for atomicity and consistency
- Add mock implementations for relay clients, chaintime service, and database setters
- Add comprehensive unit tests covering bid handling, metadata management, parameter validation, and service initialization

## Motivation

This work ensures that start slot metadata updates are handled transactionally, maintaining consistency across relay providers, and provides test coverage for core bids service functionality.

## Testing

All new tests pass and validate the implementation across multiple scenarios including success cases, error handling, and edge cases.

### CODE REVIEW

File: services/bids/standard/service_test.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
+// Copyright © 2022, 2024 Weald Technology Trading.
```
Comment (Bez625) [inline services/bids/standard/service_test.go]: Please update the year in the copyright. New files should be 2026

File: services/bids/standard/metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
+// Copyright © 2022, 2024 Weald Technology Trading.
```
Comment (Bez625) [inline services/bids/standard/metadata_test.go]: Again please update the copyright.

File: services/bids/standard/parameters_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
+// Copyright © 2022, 2024 Weald Technology Trading.
```
Comment (Bez625) [inline services/bids/standard/parameters_test.go]: Again please update the copyright.

File: services/bids/standard/service.go [hunk 4]
```diff
@@ -110,6 +114,34 @@ func (s *Service) onStart(ctx context.Context,
 	return nil
 }
 
+func updateStartSlot(ctx context.Context, s *Service, md *metadata, startSlot int64) error {
```
Comment (Bez625) [inline services/bids/standard/service.go]: Is there a reason you have passed in the `s *Service` as a parameter rather than making this new function as a method on the struct directly? Given the function heavily relies on the struct internals I think it might make more sense to make it a method of the Bid service.

**Review by Bez625 — CHANGES_REQUESTED** [review]
Nice tests, some small updates requested, but looking good.

File: services/bids/standard/service.go (hunk 4 above)
Comment (xav) [reply services/bids/standard/service.go]: I guess I just refactored the code out and didn't think all the way. Changed to struct member instead.
